src/any_agent/frameworks/openai.py

```diff
@@ -45,6 +52,22 @@ def __init__(
     def framework(self) -> AgentFramework:
         return AgentFramework.OPENAI
 
+    def _wrap_tools(self, tools: list[ToolFunction[...]]) -> list[Tool]:
+        wrapped_tools: list[Tool] = [
+            function_tool(
+                tool, failure_error_function=self.any_agent_tool_error_function
+            )
+            for tool in tools
+        ]
+        return wrapped_tools
+
+    @staticmethod
+    def any_agent_tool_error_function(
+        ctx: RunContextWrapper[Any], error: Exception
+    ) -> str:
+        traceback.print_exception(error)
+        return f"An error occurred while running the tool. Please try again. Error: {error!s}"
```

Comment:
If the existing error messages aren't sufficient, should we suggest this as an improvement to the agent SDK itself vs adding a wrapper in our library? This seems like we're editing the functionality of the framework itself

Reply:
The agent SDK provides this function as a default option. It should be the user's choice to use something else if needed.